MVPInternshipApp.Server/Controllers/CustomerController.cs

```diff
@@ -22,14 +18,15 @@ public CustomerController(SalesprojectContext context)
 
         // GET: api/Customer
         [HttpGet]
-        public async Task<ActionResult<IEnumerable<Customer>>> GetCustomers()
+        public async Task<ActionResult<IEnumerable<CustomerDto>>> GetCustomers()
         {
-            return await _context.Customers.ToListAsync();
+            var customers = await _context.Customers.ToListAsync();
+            return Ok(customers.Select(x => new CustomerDto(x)));
         }
 
         // GET: api/Customer/5
         [HttpGet("{id}")]
-        public async Task<ActionResult<Customer>> GetCustomer(int id)
+        public async Task<ActionResult<CustomerDto>> GetCustomer(int id)
         {
             var customer = await _context.Customers.FindAsync(id);
```

Comment:
Before directly looking if the ID is in the records or not, can you please verify first if the ID's value is a valid content? This way, you can be able to save your resource and reduce cost.